api/src/DataProvider/TopBookDataProvider.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataProvider;
+
+use App\Entity\TopBook;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+final class TopBookDataProvider
+{
+    private const MAX_CACHE_TIME = 3600; // 1 hour
+    private const DATA_SOURCE = 'top-100-novel-sci-fi-fr.csv';
+    private const FIELDS_COUNT = 5;
+
+    private CacheInterface $cache;
+    private KernelInterface $kernel;
+
+    public function __construct(CacheInterface $cache, KernelInterface $kernel)
+    {
+        $this->cache = $cache;
+        $this->kernel = $kernel;
+    }
+
+    /**
+     * Local caching is done so the CSV isn't reloaded at every call.
+     */
+    public function getTopBooks(): array
+    {
+        return $this->cache->get('books.sci-fi.top.fr', function (ItemInterface $item) {
```

Comment:
For debugging, I recommend you to have 2 classes: TopBookDataRepository, which doesn't handle any cache; and TopBookCachedDataRepository, which decorates the first one to add cache on it.

Tip: use an interface to respect SOLID principles.